assets/js/components/ChargingPlanRepetitiveSettingsEntries.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+	<div v-for="(plan, index) in plans" :key="index">
+		<ChargingPlanRepetitiveSettingsEntry
+			class="mb-4"
+			:id="index"
+			v-bind="plan"
+			:socBasedPlanning="socBasedPlanning"
+			:rangePerSoc="rangePerSoc"
+			@repetitive-plan-updated="updateRepetitivePlan"
+			@repetitive-plan-removed="removeRepetitivePlan"
+		/>
+	</div>
+	<div class="d-flex align-items-baseline">
+		<button
+			type="button"
+			class="d-flex btn btn-sm btn-outline-primary border-0 ps-0 align-items-baseline"
+			data-testid="repetitive-plan-add"
+			@click="addRepetitivePlan"
+		>
+			<shopicon-regular-plus size="s" class="flex-shrink-0 me-2"></shopicon-regular-plus>
+			<p class="mb-0">{{ $t("main.chargingPlan.addRepetitivePlan") }}</p>
+		</button>
+		<button
+			v-if="hasDataChanged"
+			type="button"
+			class="btn btn-sm btn-outline-primary ms-3 border-0 text-decoration-underline"
+			data-testid="plan-apply"
+			@click="update"
+		>
+			{{ $t("main.chargingPlan.update") }}
+		</button>
+	</div>
+</template>
+
+<script>
+import api from "../api";
+import ChargingPlanRepetitiveSettingsEntry from "./ChargingPlanRepetitiveSettingsEntry.vue";
+
+const DEFAULT_WEEKDAYS = [0];
+const DEFAULT_TARGET_TIME = "12:00";
+const DEFAULT_TARGET_SOC = 80;
+
+export default {
+	name: "ChargingPlanRepetitiveSettingsEntries",
+	components: {
+		ChargingPlanRepetitiveSettingsEntry,
+	},
+	props: {
+		id: Number,
+		socBasedPlanning: Boolean,
+		rangePerSoc: Number,
+	},
+	data: function () {
+		return {
+			plans: [],
+			initialPlans: [],
+		};
+	},
+	mounted() {
+		this.fetchRepetitivePlans();
+	},
+	computed: {
+		hasDataChanged: function () {
+			return JSON.stringify(this.initialPlans) !== JSON.stringify(this.plans);
+		},
+	},
+	methods: {
+		fetchRepetitivePlans: async function () {
+			let response = await api.get(`/loadpoints/${this.id}/plan/repetitive`);
```

Comment:
Die Komponente sollte sich nicht selbst die Daten holen. Das Muster was wir standardmäßig verwenden ist, dass wir Schreiboperationen via REST machen die aktualisierten Daten aber über Websockets gepusht werden. Die landen dann im globalen State. Da dieses Feature nur für Fahrzeuge mit SoC verfügbar sein wird sollten wir den Ladeplan auch direkt am Fahrzeug pushen. Siehe hier:

![Bildschirmfoto 2024-09-16 um 08 35 18](https://github.com/user-attachments/assets/11adb28a-b9ee-46bb-812d-be6717478a5e)

Damit ist er dann auch gleich über MQTT verfügbar und wir lösen das Problem Datensync zwischen Geräten. Also mehrere offene Browserfenster.

Aktuell veröffentlichen wir den "einmaligen Plan" auch noch nicht am Fahrzeug. Das müssen wir aber in diesem Zuge mit machen. Momentan kommen die aktuellen Plandaten die für die Statusanzeige in der UI und für das Ladeplanformular verwendet werden vom Ladepunkt. Mit der Einführung von mehreren Plänen würde ich am Ladepunkt nur noch die Daten des "als nächstes anstehenden Plans" veröffentlichen. Dass kann ein einmaliger sein, aber auch eine Ladung, die aufgrund einer Wiederholungsregel ansteht. Bislang mussten wir das nicht trennen. Jetzt ist das aber erforderlich.

Reply:
Beim Vehicle werden zwei Arrays mitgegeben: `plans` und `repeatingPlans`.
Beim Loadpoint werden die `effectivePlanSoc` und `effectivePlanTime`-Felder mitgeschickt.
Passt das so?